src/cloud/components/Modal/contents/Dashboard/CreateDashboardModal.tsx

```diff
@@ -15,29 +16,28 @@ interface CreateDashboardModalProps {
 const CreateDashboardModal = ({ onCreate }: CreateDashboardModalProps) => {
   const { closeLastModal: closeModal } = useModal()
   const { team } = usePage()
-  const { createDashboardApi } = useCloudApi()
+  const { createDashboardApi, createViewApi } = useCloudApi()
   const [sending, setSending] = useState(false)
-  const { push } = useRouter()
 
   const submit = useCallback(
     async (body: CreateDashboardRequestBody) => {
       if (team == null) {
         return
       }
       setSending(true)
-      await createDashboardApi(team.id, body, {
-        afterSuccess: (dashboard) => {
-          closeModal()
-          if (onCreate != null) {
-            return onCreate(dashboard)
-          } else {
-            push(getTeamLinkHref(team, 'index', { dashboard: dashboard.id }))
-          }
-        },
-      })
+      const res = await createDashboardApi(team.id, body)
+
+      if (!res.err) {
+        const { data: dashboard } = res.data as CreateDashboardResponseBody
+        await createViewApi({ dashboard: dashboard.id, type: 'table' })
```

Comment:
Should handle error

Reply:
all the current api calls already handle errors by default

```
 const createViewApi = useCallback(
    async (target: CreateViewRequestBody) => {
      return send(shortid.generate(), 'create', {
        api: () => createView(target),
        cb: ({ data }: CreateViewResponseBody) => {
          updateViewsMap([data.id, data])
        },
      })
    },
    [updateViewsMap, send]
  )


const send = useCallback(
    async (id: string, act: string, { api, cb }) => {
      const res = { err: false, error: undefined, data: undefined }
      if (sendingMap.get(id)) {
        return
      }

      setSendingMap((prev) => {
        const newMap = new Map(prev)
        newMap.delete(id)
        newMap.set(id, act)
        return newMap
      })
      try {
        const data = await api()
        res.data = data
        if (cb != null) {
          cb(data)
        }
      } catch (error) {
        res.err = true
        res.error = error
        console.error(error)
        pushApiErrorMessage(error)
      }

      setSendingMap((prev) => {
        const newMap = new Map(prev)
        newMap.delete(id)
        return newMap
      })

      return res
    },
    [sendingMap, pushApiErrorMessage]
  )

```